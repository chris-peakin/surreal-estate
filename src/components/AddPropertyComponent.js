import React, { useState } from "react";
import "../styles/AddPropertyStyles.css";
import Alert from "./AlertComponent";

const axios = require("axios").default;

function AddProperty() {
  const initialState = {
    fields: {
      title: "",
      city: "Leeds",
      type: "Terraced",
      bedrooms: "",
      bathrooms: "",
      price: "",
      email: "",
    },
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const [alert, setAlert] = useState(initialState.alert);
  const [fields, setFields] = useState(initialState.fields);

  function handleAddProperty(event) {
    event.preventDefault();
    setAlert({ message: "", isSuccess: false });
    axios
      .post("http://localhost:4000/api/v1/PropertyListing/", {
        title: fields.title,
        city: fields.city,
        type: fields.type,
        bedrooms: parseInt(fields.bedrooms, 10),
        bathrooms: parseInt(fields.bathrooms, 10),
        price: parseInt(fields.price, 10),
        email: fields.email,
      })
      .then(() => setAlert({ message: "Property Added", isSuccess: true }))
      .catch(() =>
        setAlert({
          message: "Server error. Please try again later.",
          isSuccess: false,
        })
      );
  }

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="AddProperty">
      <form onSubmit={handleAddProperty} className="Form">
        <Alert message={alert.message} success={alert.isSuccess} />
        <div className="FormRule">
          <input
            id="title"
            name="title"
            value={fields.title}
            onChange={handleFieldChange}
            placeholder="property title"
          />
        </div>
        <div className="FormRule">
          <label htmlFor="city">
            <select
              id="city"
              name="city"
              value={fields.city}
              onChange={handleFieldChange}
              placeholder="city"
            >
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Sheffield">Sheffield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>
          <label htmlFor="type">
            <select
              id="type"
              name="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-Detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Bungalow">Bungalow</option>
              <option value="Flat">Flat</option>
            </select>
          </label>
        </div>
        <div className="FormRule">
          <input
            id="bedrooms"
            name="bedrooms"
            value={fields.bedrooms}
            onChange={handleFieldChange}
            placeholder="no. of bedrooms"
          />
          <input
            id="bathrooms"
            name="bathrooms"
            value={fields.bathrooms}
            onChange={handleFieldChange}
            placeholder="no. of bathrooms"
          />
          <input
            id="price"
            name="price"
            value={fields.price}
            onChange={handleFieldChange}
            placeholder="property price"
          />
        </div>
        <div className="FormRule">
          <input
            id="email"
            name="email"
            value={fields.email}
            onChange={handleFieldChange}
            placeholder="your email eg. joe.bloggs@gmail.com"
          />
        </div>
        <button type="submit">Add Your Property</button>
      </form>
    </div>
  );
}

export default AddProperty;
