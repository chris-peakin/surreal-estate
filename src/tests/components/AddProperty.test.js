import React from "react";
import renderer from "react-test-renderer";
import AddPropertyComponent from "../../components/AddPropertyComponent";

test("add properties component renders correctly", () => {
  const testRenderer = renderer.create(<AddPropertyComponent />);
  expect(testRenderer).toMatchSnapshot();
});
