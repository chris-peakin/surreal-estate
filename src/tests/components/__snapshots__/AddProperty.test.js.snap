// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add properties component renders correctly 1`] = `
<div
  className="AddProperty"
>
  <form
    className="Form"
    onSubmit={[Function]}
  >
    <div
      className="FormRule"
    >
      <input
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="property title"
        value=""
      />
    </div>
    <div
      className="FormRule"
    >
      <label
        htmlFor="city"
      >
        <select
          id="city"
          name="city"
          onChange={[Function]}
          placeholder="city"
          value="Leeds"
        >
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Sheffield"
          >
            Sheffield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </label>
      <label
        htmlFor="type"
      >
        <select
          id="type"
          name="type"
          onChange={[Function]}
          value="Terraced"
        >
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-Detached
          </option>
          <option
            value="Terraced"
          >
            Terraced
          </option>
          <option
            value="End of Terrace"
          >
            End of Terrace
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
          <option
            value="Flat"
          >
            Flat
          </option>
        </select>
      </label>
    </div>
    <div
      className="FormRule"
    >
      <input
        id="bedrooms"
        name="bedrooms"
        onChange={[Function]}
        placeholder="no. of bedrooms"
        value=""
      />
      <input
        id="bathrooms"
        name="bathrooms"
        onChange={[Function]}
        placeholder="no. of bathrooms"
        value=""
      />
      <input
        id="price"
        name="price"
        onChange={[Function]}
        placeholder="property price"
        value=""
      />
    </div>
    <div
      className="FormRule"
    >
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="your email eg. joe.bloggs@gmail.com"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Add Your Property
    </button>
  </form>
</div>
`;
